src/Flysystem/Adapter/FedoraAdapter.php

```diff
@@ -253,7 +253,8 @@ public function write($path, $contents, Config $config) {
     $headers = [
       'Content-Type' => $this->mimeTypeGuesser->guess($path),
     ];
-    if ($this->has($path)) {
+    $heads_response = $this->fedora->getResourceHeaders($path);
+    if (isset($heads_response) && $heads_response->getStatusCode() == 200) {
```

Comment:
This seems the same as what is occuring in the `has()` function. Just wondering why you needed to pull it out here?

Reply:
I did that when i was fixing the tests because i was getting a null $response  - swapped it back